sdk/protocol-manager-poc/src/implementation/makerPlugin.ts

```diff
@@ -0,0 +1,221 @@
+                    token: collateralToken,
+                    // TODO: quote the OSM, we need to trick the contract that is SPOT that is doing the query (from in tx is SPOT)
+                    price: await ctx.priceService.getPrice({baseToken: collateralToken, quoteToken }),
+                    nextPrice: Price.createFrom({ value: spotRes.liquidationRatio.toString(), baseToken: collateralToken, quoteToken: quoteToken }), // TODO
+                    priceUSD: await ctx.priceService.getPriceUSD(collateralToken),
+
+                    // For Maker these fields are the same
+                    liquidationThreshold: RiskRatio.createFrom({ ratio: Percentage.createFrom({ percentage: spotRes.liquidationRatio.times(100).toNumber() }), type: RiskRatio.type.CollateralizationRatio }),
+                    maxLtv: RiskRatio.createFrom({ ratio: Percentage.createFrom({ percentage: spotRes.liquidationRatio.times(100).toNumber() }), type: RiskRatio.type.CollateralizationRatio }),
+
+                    tokensLocked: TokenAmount.createFrom({token: collateralToken, amount: '0'}), // TODO check the gem balance of join adapter
+                    maxSupply: TokenAmount.createFrom({token: collateralToken, amount: Number.MAX_SAFE_INTEGER.toString()}),
+                    liquidationPenalty: Percentage.createFrom({ percentage: dogRes.liquidationPenalty.toNumber() }),
+                    // apy: Percentage.createFrom({ percentage: 0 }),
+                }
+            }
+
+            const debts = {
+                [quoteToken.address.value]: {
+                    token: quoteToken,
+                    price: await ctx.priceService.getPrice({baseToken: quoteToken, quoteToken: collateralToken }),
+                    priceUSD: await ctx.priceService.getPriceUSD(quoteToken),
+                    rate: Percentage.createFrom({ percentage: stabilityFee.times(100).toNumber() }),
+                    totalBorrowed: TokenAmount.createFrom({token: quoteToken, amount: vatRes.normalizedIlkDebt.times(vatRes.debtScalingFactor).toString()}),
+                    debtCeiling: TokenAmount.createFrom({token: quoteToken, amount: vatRes.debtCeiling.toString()}),
+                    debtAvailable: TokenAmount.createFrom({token: quoteToken, amount:  vatRes.debtCeiling.minus(vatRes.normalizedIlkDebt.times(vatRes.debtScalingFactor)).toString()}),
+                    dustLimit: TokenAmount.createFrom({token: quoteToken, amount: vatRes.debtFloor.toString()}),
+                    originationFee: Percentage.createFrom({ percentage: 0 })
+                }
+            }
+
+            return {
+                type: PoolType.Lending,
+                poolId: makerPoolId,
+                protocol: makerPoolId.protocol,
+                baseCurrency: poolBaseCurrencyToken,
+                collaterals,
+                debts
+            }
+        },
+        getPositionId: (positionId: string): IPositionId => {
+            return positionId as IPositionId
```

Comment:
are we using that somewhere? perhaps worth to make it not implemented instead?
Sorry but I don't remember what was the agreement on that one